data-processing-lib/python/src/data_processing/runtime/pure_python/transform_orchestrator.py

```diff
@@ -54,34 +55,40 @@ def orchestrate(
         if len(files) == 0:
             logger.error("No input files to process - exiting")
             return 0
+        if retries > 0:
+            statistics.add_stats({"data access retries": retries})
         logger.info(f"Number of files is {len(files)}, source profile {profile}")
         # Print interval
         print_interval = int(len(files) / 100)
         if print_interval == 0:
             print_interval = 1
-        if retries > 0:
-            statistics.add_stats({"data access retries": retries})
-        # create executor
-        executor = PythonTransformFileProcessor(
-            data_access_factory=data_access_factory, statistics=statistics, runtime_configuration=runtime_config
-        )
-        # process data
-        logger.debug(f"{runtime_config.get_name()} Begin processing files")
-        t_start = time.time()
-        completed = 0
-        for path in files:
-            executor.process_file(path)
-            completed += 1
-            if completed % print_interval == 0:
-                logger.info(
-                    f"Completed {completed} files ({100 * completed / len(files)}%) "
-                    f"in {(time.time() - t_start)/60} min"
-                )
-        logger.debug(f"Done processing {completed} files, waiting for flush() completion.")
-        # invoke flush to ensure that all results are returned
-        start = time.time()
-        executor.flush()
-        logger.info(f"done flushing in {time.time() - start} sec")
+        if execution_config.multiprocessing:
+            # using multiprocessor pool for execution
+            statistics = process_transforms(files=files, size=execution_config.num_processors,
```

Comment:
this is asymetric - in one case there is a single method and the other case a class plus lots of code.  What if we add a `process_files(...,files:list[string]...)` method to the `AbstractTransformFileProcessor` class and then have different implementations for serial and parallel?

Reply:
Actually maybe we need an `AbstractPythonTransformFileProcessor` that adds this `process_files()` method and then two implementations for the two sub-classes?

Reply:
changed layout

Reply:
i was not suggesting a formatting change :)